src/Tmds.ExecFunction/ExecFunction.cs

```diff
@@ -282,12 +282,109 @@ private static string[] GetApplicationArguments()
                 string commandLine = Marshal.PtrToStringAuto(ptr);
                 return CommandLineToArgs(commandLine);
             }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

Comment:
Can you move the OSX code block into a separate function?

Reply:
Moved it into a separate function: `GetOSXCommandLineArguments`